src/metatensor/models/utils/data/readers/targets/ase.py

```diff
@@ -1,43 +1,157 @@
+    filename: str,
+    key: str,
+) -> TensorBlock:
+    """Store force information as a :class:`metatensor.TensorBlock` which can be used as
+    ``position`` gradients .
 
-    if type(target_values) is str:
-        target_values = [target_values]
+    :param filename: name of the file to read
+    :param key: target value key name to be parsed from the file.
 
+    :returns:
+        TensorMap containing the given information
+    """
     frames = ase.io.read(filename, ":")
+    n_structures = len(frames)
+    # We store forces as positions gradients which means we invert the sign
+    values_raw = [-torch.tensor(f.arrays[key]) for f in frames]
 
-    target_dictionary = {}
-    for target_value in target_values:
-        values = [f.info[target_value] for f in frames]
+    # The `"sample"` label refers to the index of the corresponding value in the
+    # block. Here, the number of values is the same as the number of structures so
+    # we can keep `"sample"` and `"structure"` the same.
+    samples = Labels(
+        ["sample", "structure", "atom"],
+        torch.tensor(
+            [[s, s, a] for s in range(n_structures) for a in range(len(values_raw[s]))]
+        ),
+    )
 
-        n_structures = len(values)
+    values = torch.concatenate(values_raw, dim=0)
+    assert values.shape[1] != (3,)
 
-        block = TensorBlock(
-            values=torch.tensor(values).reshape(-1, 1),
-            samples=Labels(["structure"], torch.arange(n_structures).reshape(-1, 1)),
-            components=[],
-            properties=Labels.single(),
-        )
+    block = TensorBlock(
+        values=values.reshape(-1, 3, 1),
+        samples=samples,
+        components=[Labels(["direction"], torch.arange(3).reshape(-1, 1))],
+        properties=Labels.single(),
+    )
+
+    return block
 
-        target_dictionary[target_value] = TensorMap(
-            keys=Labels(["lambda", "sigma"], torch.tensor([(0, 1)])), blocks=[block]
+
+def read_virial_ase(
+    filename: str,
+    key: str,
+):
+    """Store virial information in :class:`metatensor.TensorBlock` which can be used as
+    ``displacement`` gradients
+
+    :param filename: name of the file to read
+    :param key: target value key name to be parsed from the file
+
+    :returns:
+        TensorMap containing the given information
+    """
+    return _read_virial_stress_ase(filename=filename, key=key, is_virial=True)
+
+
+def read_stress_ase(
+    filename: str,
+    key: str,
+):
+    """Store stress information in :class:`metatensor.TensorBlock` which can be used as
+    ``displacement`` gradients
+
+    :param filename: name of the file to read
+    :param key: target value key name to be parsed from the file
+
+    :returns:
+        TensorMap containing the given information
+    """
+    return _read_virial_stress_ase(filename=filename, key=key, is_virial=False)
+
+
+def _read_virial_stress_ase(
+    filename: str,
+    key: str,
+    is_virial: bool = True,
```

Comment:
I don't know if this function should have a default for this argument

Reply:
See your point. For energy we could set it to `"energy"`, `"forces"` `"virial"` and `"stress"`. But, for this private function I would leave it mandatory.